packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -0,0 +1,46 @@
+import { Request, Response } from 'express';
+import { Giveaway } from '../models/giveaway.model';
+import { giveawaysContract } from '../contracts';
+
+export const getGiveaways = async (req: Request, res: Response) => {
+  try {
+    const giveaways = await Giveaway.find();
+    res.status(200).json(giveaways);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+};
+
+export const getGiveaway = async (req: Request, res: Response) => {
+  try {
+    const id = req.params.id;
+    const giveaway = await Giveaway.findById(id);
+    return res.status(200).json(giveaway);
+  } catch (error) {
+    return res.status(500).json(error);
+  }
+};
+  
+export const createGiveaway = async (req: Request, res: Response) => {
+  try {
+    const { title, description, startTime, endTime, numberOfWinners, requirements } = req.body;
+    const giveaway = new Giveaway({
+      title,
+      description,
+      startTime: new Date(startTime),
+      endTime: new Date(endTime),
+      numberOfWinners,
+      requirements
+    });
```

Comment:
Shouldn't we perform some validations on the input? For example, the end time > start time, and negative number of winners. I know that this validation is in the contract, but in these case I think it should return a bad request instead